diff -ruN ./original/binutils-2.24/bfd/config.bfd ./binutils-2.24/bfd/config.bfd
--- ./original/binutils-2.24/bfd/config.bfd	2013-11-04 10:33:37.000000000 -0500
+++ ./binutils-2.24/bfd/config.bfd	2014-05-28 18:00:56.000000000 -0400
@@ -520,6 +520,10 @@
     targ64_selvecs="bfd_elf64_x86_64_sol2_vec bfd_elf64_l1om_vec bfd_elf64_k1om_vec"
     want64=true
     ;;
+  i[3-7]86-*-stewieos*)
+    targ_defvec=bfd_elf32_i386_vec
+    targ_selvecs=i386coff_vec
+    ;;
 #ifdef BFD64
   x86_64-*-solaris2*)
     targ_defvec=bfd_elf32_i386_sol2_vec
diff -ruN ./original/binutils-2.24/config.sub ./binutils-2.24/config.sub
--- ./original/binutils-2.24/config.sub	2013-11-26 06:37:33.000000000 -0500
+++ ./binutils-2.24/config.sub	2014-05-28 17:57:18.000000000 -0400
@@ -1356,7 +1356,7 @@
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -stewieos* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -ruN ./original/binutils-2.24/gas/configure.tgt ./binutils-2.24/gas/configure.tgt
--- ./original/binutils-2.24/gas/configure.tgt	2013-11-04 10:33:37.000000000 -0500
+++ ./binutils-2.24/gas/configure.tgt	2014-05-28 18:02:08.000000000 -0400
@@ -245,6 +245,7 @@
   i386-*-isc*)				fmt=coff ;;
   i386-*-sco3.2v5*)			fmt=elf ;;
   i386-*-sco3.2*)			fmt=coff ;;
+  i386-*-stewieos*)			fmt=elf ;;
   i386-*-vsta)				fmt=aout ;;
   i386-*-msdosdjgpp* \
   | i386-*-go32*)			fmt=coff em=go32 ;;
diff -ruN ./original/binutils-2.24/ld/configure.tgt ./binutils-2.24/ld/configure.tgt
--- ./original/binutils-2.24/ld/configure.tgt	2013-11-26 06:37:33.000000000 -0500
+++ ./binutils-2.24/ld/configure.tgt	2014-05-28 18:03:22.000000000 -0400
@@ -220,6 +220,7 @@
 i[3-7]86-*-aros*)	targ_emul=elf_i386 ;;
 i[3-7]86-*-rdos*)	targ_emul=elf_i386 ;;
 x86_64-*-rdos*)		targ_emul=elf64rdos ;;
+i[3-7]86-*-stewieos*)	targ_emul=stewieos_i386 ;;
 i[3-7]86-*-bsd)		targ_emul=i386bsd ;;
 i[3-7]86-*-bsd386)	targ_emul=i386bsd ;;
 i[3-7]86-*-bsdi*)	targ_emul=i386bsd ;;
diff -ruN ./original/binutils-2.24/ld/emulparams/stewieos_i386.sh ./binutils-2.24/ld/emulparams/stewieos_i386.sh
--- ./original/binutils-2.24/ld/emulparams/stewieos_i386.sh	1969-12-31 19:00:00.000000000 -0500
+++ ./binutils-2.24/ld/emulparams/stewieos_i386.sh	2014-05-28 18:07:24.000000000 -0400
@@ -0,0 +1,15 @@
+SCRIPT_NAME=elf
+OUTPUT_FORMAT=elf32-i386
+TEXT_START_ADDR=0x100000
+MAXPAGESIZE="CONSTANT (MAXPAGESIZE)"
+COMMONPAGESIZE="CONSTANT (COMMONPAGESIZE)"
+
+ARCH=i386
+MACHINE=
+NOP=0x90909090
+
+TEMPLATE_NAME=elf32
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
+NO_SMALL_DATA=yes
+SEPARATE_GOTPLT=12
\ No newline at end of file
diff -ruN ./original/binutils-2.24/ld/Makefile.in ./binutils-2.24/ld/Makefile.in
--- ./original/binutils-2.24/ld/Makefile.in	2013-11-26 06:37:33.000000000 -0500
+++ ./binutils-2.24/ld/Makefile.in	2014-05-28 18:11:20.000000000 -0400
@@ -571,6 +571,7 @@
 	eelf32xstormy16.c \
 	eelf32xtensa.c \
 	eelf_i386.c \
+	estewieos_i386.c \
 	eelf_i386_be.c \
 	eelf_i386_chaos.c \
 	eelf_i386_fbsd.c \
@@ -2752,6 +2753,8 @@
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386 "$(tdir_elf_i386)"
+estewieos_i386.c: $(srcdir)/emulparams/stewieos_i386.sh $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} stewieos_i386 "$(tdir_stewieos_i386)"
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_be "$(tdir_elf_i386_be)"
