diff -ruN ./original/gcc-4.9.0/config.sub ./gcc-4.9.0/config.sub
--- ./original/gcc-4.9.0/config.sub	2013-10-01 12:50:56.000000000 -0400
+++ ./gcc-4.9.0/config.sub	2014-05-28 20:36:48.000000000 -0400
@@ -1356,7 +1356,7 @@
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -stewieos* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -ruN ./original/gcc-4.9.0/gcc/config/stewieos.h ./gcc-4.9.0/gcc/config/stewieos.h
--- ./original/gcc-4.9.0/gcc/config/stewieos.h	1969-12-31 19:00:00.000000000 -0500
+++ ./gcc-4.9.0/gcc/config/stewieos.h	2014-05-28 20:42:32.000000000 -0400
@@ -0,0 +1,8 @@
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()		\
+  do {						\
+  	builtin_define_std("stewieos");		\
+  	builtin_define_std("unix");		\
+  	builtin_assert("system=stewieos");	\
+  	builtin_assert("system=unix");		\
+  } while(0);
diff -ruN ./original/gcc-4.9.0/gcc/config.gcc ./gcc-4.9.0/gcc/config.gcc
--- ./original/gcc-4.9.0/gcc/config.gcc	2014-03-12 06:13:07.000000000 -0400
+++ ./gcc-4.9.0/gcc/config.gcc	2014-05-28 18:14:56.000000000 -0400
@@ -643,6 +643,12 @@
     "" | yes | posix) thread_file='posix' ;;
   esac
   ;;
+*-*-stewieos*)
+  extra_parts="crtbegin.o crtend.o"
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  ;;
 *-*-freebsd*)
   # This is the generic ELF configuration of FreeBSD.  Later
   # machine-specific sections may refine and add to this
@@ -875,6 +881,11 @@
 esac
 
 case ${target} in
+i[3-7]86-*-stewieos*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h stewieos.h"
+	tmake_file="i386/t-i386elf t-svr4"
+	use_fixproto=yes
+	;;
 aarch64*-*-elf)
 	tm_file="${tm_file} dbxelf.h elfos.h newlib-stdint.h"
 	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-elf-raw.h"
diff -ruN ./original/gcc-4.9.0/libgcc/config.host ./gcc-4.9.0/libgcc/config.host
--- ./original/gcc-4.9.0/libgcc/config.host	2014-03-27 11:40:31.000000000 -0400
+++ ./gcc-4.9.0/libgcc/config.host	2014-05-28 18:20:18.000000000 -0400
@@ -1230,6 +1230,10 @@
 	tmake_file="mep/t-mep t-fdpbit"
 	extra_parts="crtbegin.o crtend.o"
 	;;
+i[3-7]86-*-stewieos*)
+	extra_parts="crtbegin.o crtend.o"
+	tmake_file="$tmake_file i386/t-crtstuff"
+	;;
 *)
 	echo "*** Configuration ${host} not supported" 1>&2
 	exit 1
